website/source/api/features/filtering.html.md

```diff
@@ -0,0 +1,458 @@
+    }
+}
+```
+
+#### Catalog API
+
+**Command - Unfiltered**
+
+```sh
+curl -X GET localhost:8500/v1/catalog/service/api-internal
+```
+
+**Response - Unfiltered**
+
+```json
+[
+    {
+        "ID": "b4f64e8c-5c7d-11e9-bf68-8c8590bd0966",
+        "Node": "node-1",
+        "Address": "198.18.0.1",
+        "Datacenter": "dc1",
+        "TaggedAddresses": null,
+        "NodeMeta": {
+            "agent": "true",
+            "arch": "i386",
+            "os": "darwin"
+        },
+        "ServiceKind": "",
+        "ServiceID": "api-internal",
+        "ServiceName": "api-internal",
+        "ServiceTags": [
+            "tag"
+        ],
+        "ServiceAddress": "",
+        "ServiceWeights": {
+            "Passing": 1,
+            "Warning": 1
+        },
+        "ServiceMeta": {
+            "environment": "qa"
+        },
+        "ServicePort": 9090,
+        "ServiceEnableTagOverride": false,
+        "ServiceProxyDestination": "",
+        "ServiceProxy": {},
+        "ServiceConnect": {},
+        "CreateIndex": 30,
+        "ModifyIndex": 30
+    },
+    {
+        "ID": "b4faf93a-5c7d-11e9-840d-8c8590bd0966",
+        "Node": "node-2",
+        "Address": "198.18.0.2",
+        "Datacenter": "dc1",
+        "TaggedAddresses": null,
+        "NodeMeta": {
+            "arch": "arm",
+            "os": "linux"
+        },
+        "ServiceKind": "",
+        "ServiceID": "api-internal",
+        "ServiceName": "api-internal",
+        "ServiceTags": [
+            "test",
+            "tag"
+        ],
+        "ServiceAddress": "",
+        "ServiceWeights": {
+            "Passing": 1,
+            "Warning": 1
+        },
+        "ServiceMeta": {
+            "environment": "production"
+        },
+        "ServicePort": 9090,
+        "ServiceEnableTagOverride": false,
+        "ServiceProxyDestination": "",
+        "ServiceProxy": {},
+        "ServiceConnect": {},
+        "CreateIndex": 29,
+        "ModifyIndex": 29
+    },
+    {
+        "ID": "b4fbe7f4-5c7d-11e9-ac82-8c8590bd0966",
+        "Node": "node-4",
+        "Address": "198.18.0.4",
+        "Datacenter": "dc1",
+        "TaggedAddresses": null,
+        "NodeMeta": {
+            "arch": "i386",
+            "os": "freebsd"
+        },
+        "ServiceKind": "",
+        "ServiceID": "api-internal",
+        "ServiceName": "api-internal",
+        "ServiceTags": [],
+        "ServiceAddress": "",
+        "ServiceWeights": {
+            "Passing": 1,
+            "Warning": 1
+        },
+        "ServiceMeta": {
+            "environment": "qa"
+        },
+        "ServicePort": 9090,
+        "ServiceEnableTagOverride": false,
+        "ServiceProxyDestination": "",
+        "ServiceProxy": {},
+        "ServiceConnect": {},
+        "CreateIndex": 28,
+        "ModifyIndex": 28
+    }
+]
+```
+
+**Command - Filtered**
+
+```sh
+curl -G localhost:8500/v1/catalog/service/api-internal --data-urlencode 'filter=NodeMeta.os == linux'
```

Comment:
This is another internal endpoint example that we perhaps should omit.

Reply:
Same as before, thats a service name not an actual internal endpoint.